pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTAnonymousClass.java

```diff
@@ -4,14 +4,15 @@
 
 package net.sourceforge.pmd.lang.apex.ast;
 
-import com.google.summit.ast.Node;
 import net.sourceforge.pmd.annotation.InternalApi;
 
-public class ASTAnonymousClass extends ApexRootNode<Node> {
+import com.google.summit.ast.declaration.TypeDeclaration;
+
+public class ASTAnonymousClass extends ApexRootNode<TypeDeclaration> {
 
     @Deprecated
     @InternalApi
-    public ASTAnonymousClass(Node anonymousClass) {
+    public ASTAnonymousClass(TypeDeclaration anonymousClass) {
```

Comment:
Was the more generic Node parameter because of the factory construction in the tree builder? Anyway, I agree that it is better to use the more specific type.  Thanks.

Reply:
I initially set all the type arguments as `Node` during #1. A more specific type argument is only needed when a wrapper class's public methods need access to the particular properties of the underlying node, e.g. `ASTUserClass` needs `ClassDeclaration.extendsType`. Of course, even wrapper classes with no additional methods should have an appropriate type argument if they directly map to a Summit node. I'll be changing these type arguments as needed while implementing these methods in upcoming commits.

In this case, though, I had to make this change because I changed the bound on `ApexRootNode` from `T extends Node` to `T extends TypeDeclaration` - not directly connected to anything, but I thought it would be appropriate.